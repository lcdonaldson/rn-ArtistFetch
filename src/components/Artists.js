import React from 'react';

import {
  StyleSheet,
  View,
  Text,
} from 'react-native';

const Artists = (props) => {
    return (
        <View style={styles.infoWrapper}>
            <Text style={styles.primaryText}>{props.title}</Text>
            <View style={styles.row}>
            <View style={styles.flexLeft}>
                <Text style={styles.lText}>Name:  {props.name}</Text>
                <Text style={styles.lText}>Cost:  {props.cost}</Text>
            </View>
            <View style={styles.flexRight}>
                <Text style={styles.rText}>Genre:  {props.genre}</Text>
                <Text style={styles.rText}>Released on:  {props.date}</Text>
            </View>
            </View>
        </View>
    )
}
const styles = StyleSheet.create({
  infoWrapper: {
    backgroundColor: 'lightblue',
    borderColor: 'white',
    borderWidth: 2,
    flex: 1,
    display: 'flex',
    padding: 10,
    marginBottom: 10
  },
  primaryText: {
    fontSize: 30, 
    color: '#000',
    width: '100%',
    display: 'flex',
    borderBottomColor: 'teal',
    borderBottomWidth: 2,
    marginBottom: 10
  },
  row: {
    display: 'flex',
    flexDirection: 'row',
  },
  flexLeft: {
    display: 'flex',
    flexDirection: 'column',
    width: '50%'
  },
  flexRight: {
    display: 'flex',
    flexDirection: 'column',
    width: '50%'
  },
  lText: {paddingBottom: 10, fontSize: 16},
  rText: {paddingBottom: 10, fontSize: 16}
});

export default Artists