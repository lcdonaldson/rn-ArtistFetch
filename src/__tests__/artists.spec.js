
import React from 'react';
import Artists from '../../src/components/Artists';
import renderer from 'react-test-renderer';

describe('Snapshot tests', () => {
    // SNAPSHOT
    it('renders correctly when there are no items', () => {
        const tree = renderer.create(<Artists />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});



