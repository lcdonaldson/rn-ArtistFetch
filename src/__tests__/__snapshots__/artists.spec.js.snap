// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests renders correctly when there are no items 1`] = `
<View
  style={
    Object {
      "backgroundColor": "lightblue",
      "borderColor": "white",
      "borderWidth": 2,
      "display": "flex",
      "flex": 1,
      "marginBottom": 10,
      "padding": 10,
    }
  }
>
  <Text
    style={
      Object {
        "borderBottomColor": "teal",
        "borderBottomWidth": 2,
        "color": "#000",
        "display": "flex",
        "fontSize": 30,
        "marginBottom": 10,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
          "width": "50%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
            "paddingBottom": 10,
          }
        }
      >
        Name:  
      </Text>
      <Text
        style={
          Object {
            "fontSize": 16,
            "paddingBottom": 10,
          }
        }
      >
        Cost:  
      </Text>
    </View>
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
          "width": "50%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
            "paddingBottom": 10,
          }
        }
      >
        Genre:  
      </Text>
      <Text
        style={
          Object {
            "fontSize": 16,
            "paddingBottom": 10,
          }
        }
      >
        Released on:  
      </Text>
    </View>
  </View>
</View>
`;
